substrate/client/network/sync/src/engine.rs

```diff
@@ -770,6 +782,25 @@ where
 			}
 		}
 
+		// Retreive warp sync target block header just before polling `ChainSync`
+		// to make progress as soon as we receive it.
+		if let Some(ref mut target_block_header_rx) = self.warp_sync_target_block_header_rx {
+			match target_block_header_rx.poll_unpin(cx) {
+				Poll::Ready(Ok(target)) => {
+					self.chain_sync.set_warp_sync_target_block(target);
+					self.warp_sync_target_block_header_rx = None;
+				},
+				Poll::Ready(Err(err)) => {
+					log::error!(
+						target: "sync",
```

Comment:
`LOG_TARGET` but I couldn't find the definition in the file yet

Reply:
Done